tests/test_process.py

```diff
@@ -1,124 +1,172 @@
+                    'critical': [1, 4],
+                    'warning': [2, 5]
+                }
+            },
+            'mocked_processes': 5
+        },
+        {
+            'config': {
+                'name': 'test_4',
+                'search_string': ['test_4'], # index in the array for our find_pids mock
+                'thresholds': {
+                    'critical': [1, 4],
+                    'warning': [2, 5]
+                }
+            },
+            'mocked_processes': 6
+        },
+        {
+            'config': {
+                'name': 'test_tags',
+                'search_string': ['test_5'], # index in the array for our find_pids mock
+                'tags': ['onetag', 'env:prod']
+            },
+            'mocked_processes': 1
+        },
+        {
+            'config': {
+                'name': 'test_badthresholds',
+                'search_string': ['test_6'], # index in the array for our find_pids mock
+                'thresholds': {
+                    'test': 'test'
+                }
+            },
+            'mocked_processes': 1
+        },
+    ]
+
+    PROCESS_METRIC = [
+        'system.processes.cpu.pct',
+        'system.processes.involuntary_ctx_switches',
+        'system.processes.ioread_bytes',
+        'system.processes.ioread_count',
+        'system.processes.iowrite_bytes',
+        'system.processes.iowrite_count',
+        'system.processes.mem.real',
+        'system.processes.mem.rss',
+        'system.processes.mem.vms',
+        'system.processes.number',
+        'system.processes.open_file_descriptors',
+        'system.processes.threads',
+        'system.processes.voluntary_ctx_switches'
+    ]
+
+    def mock_find_pids(self, search_string, exact_match=True, ignore_denied_access=True):
+        idx = search_string[0].split('_')[1]
+        # Use a real PID to get real metrics!
+        return [os.getpid()] * self.CONFIG_STUBS[int(idx)]['mocked_processes']
 
     def test_check(self):
-        config = {
-            'init_config': {},
-            'instances': []
+        mocks = {
+            'find_pids': self.mock_find_pids
         }
-        self.agentConfig = {
-            'version': '0.1',
-            'api_key': 'toto'
+
+        config = {
+            'instances': [stub['config'] for stub in self.CONFIG_STUBS]
         }
 
-        self.check = load_check('process', config, self.agentConfig)
+        self.run_check(config, mocks=mocks)
+
+        for stub in self.CONFIG_STUBS:
+            # Assert metrics
+            for mname in self.PROCESS_METRIC:
+                proc_name = stub['config']['name']
+                expected_tags = [proc_name, "process_name:{0}".format(proc_name)]
 
-        config = self.build_config(config)
-        self.check.find_pids = self.find_pids
+                # If a list of tags is already there, the check extends it
+                if 'tags' in stub['config']:
+                    expected_tags = stub['config']['tags']
```

Comment:
That's a tricky one, but the answer is no because the checks extends itself the instance tags list if it exists. So here it will contain **ALL the tags**...